src/services/vault/mod.rs

```diff
@@ -0,0 +1,280 @@
+pub mod client;
+
+use crate::config::VaultConfig;
+use crate::error::APIError;
+use crate::loader::new_spinner_progress_bar;
+use crate::output::colored_println;
+use crate::{config::CONFIG_FILE, error::CliError, Config as CLIConfig};
+use async_recursion::async_recursion;
+use dialoguer::{theme::ColorfulTheme, Select};
+use log::debug;
+
+use self::client::{FetchSecretsData, VaultClient};
+
+impl Default for VaultClient {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Default for Vault {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+struct ConfigWithPath {
+    config: CLIConfig,
+    path: String,
+}
+
+pub struct Vault {}
+
+impl Vault {
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    fn get_config_with_path(&self) -> Result<ConfigWithPath, CliError> {
+        let config_file = CONFIG_FILE
+            .as_ref()
+            .expect("Unable to identify user's home directory");
+
+        let config = CLIConfig::load(config_file)?;
+
+        let config_with_path = ConfigWithPath {
+            config,
+            path: config_file.to_string(),
+        };
+
+        Ok(config_with_path)
+    }
+
+    pub async fn handle_login(&self) -> Result<bool, CliError> {
+        let mut email: Option<String> = None;
+        let mut config_with_path = self.get_config_with_path().unwrap();
+
+        if let Some(vault_config) = &config_with_path.config.vault {
+            let selections = vec!["Use the existing account", "Log in with a new account"];
+
+            let selection = Select::with_theme(&ColorfulTheme::default())
+                .with_prompt(
+                    format!(
+                "You have previously logged in as \"{}\". Would you like to continue using this account?",
+                vault_config.email
+            ))
+                .default(0)
+                .items(&selections[..])
+                .interact()?;
+
+            match selection {
+                0 => {
+                    colored_println!("Continuing with \"{}\"...", vault_config.email);
+
+                    email = Some(vault_config.email.to_string())
+                }
+                1 => {
+                    let input_email: String =
+                        dialoguer::Input::with_theme(&ColorfulTheme::default())
+                            .with_prompt("Please enter your email address")
+                            .interact()?;
+
+                    email = Some(input_email.trim().to_string());
+                }
+                _ => panic!("Invalid selection"),
+            }
+        }
+
+        // If the user has not logged in before, or has chosen to log in with a new account:
+        if email.is_none() {
+            let input_email: String = dialoguer::Input::with_theme(&ColorfulTheme::default())
```

Comment:
@jk-gan @mfauzaan since each dev already have to perform `wukong init` and use there email for the Okta auth, should we also imply that the same email will be use for Vault authentication as well ? It seems redundant to me that they will need to re-enter their email again. Given that the Wukong CLI is not supporting profiles yet we can assume that each Wukong config only has 1 profile ?

Reply:
That make sense, let me refactor a bit to use the existing profile